rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/HttpResourceDownloader.java

```diff
@@ -37,28 +37,28 @@
 
 package org.ow2.proactive_grid_cloud_portal.scheduler;
 
-import org.apache.http.impl.client.DefaultHttpClient;
-import org.apache.http.impl.conn.PoolingClientConnectionManager;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+
+import javax.ws.rs.core.Response;
+
+import org.ow2.proactive.scheduling.util.http.HttpClientBuilder;
 import org.jboss.resteasy.client.jaxrs.ResteasyClient;
 import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
 import org.jboss.resteasy.client.jaxrs.ResteasyWebTarget;
 import org.jboss.resteasy.client.jaxrs.engines.ApacheHttpClient4Engine;
 
-import javax.ws.rs.core.Response;
-import java.io.IOException;
-import java.net.HttpURLConnection;
-
 public class HttpResourceDownloader {
 
     private static final Integer CONNECTION_POOL_SIZE = 5;
 
     private static ResteasyClient client;
 
     static {
-        PoolingClientConnectionManager cm = new PoolingClientConnectionManager();
-        cm.setMaxTotal(CONNECTION_POOL_SIZE);
-        cm.setDefaultMaxPerRoute(CONNECTION_POOL_SIZE);
-        ApacheHttpClient4Engine engine = new ApacheHttpClient4Engine(new DefaultHttpClient(cm));
+        ApacheHttpClient4Engine engine =
+                new ApacheHttpClient4Engine(
+                        new HttpClientBuilder().maxConnections(CONNECTION_POOL_SIZE).useSystemProperties().build());
```

Comment:
The resources are released when the response is closed. It seems the client connection pool is initialized in a static block to prevent expensive initilizations.